src/app/qgshandlebadlayers.cpp

```diff
@@ -509,3 +495,164 @@ int QgsHandleBadLayers::layerCount()
 {
   return mLayerList->rowCount();
 }
+
+QString QgsHandleBadLayers::findFile( const QString &fileName, const QString &basePath, int maxDepth, int driveMargin )
+{
+  int depth = 0;
+  QDir folder = QDir( QDir( basePath ).absolutePath() );
+  QString existingBase;
+  // find the nearest existing folder
+  while ( !folder.exists() && folder.absolutePath().count( '/' ) > driveMargin )
+  {
+    existingBase = folder.path();
+    if ( !folder.cdUp() )
```

Comment:
I'm unsure, but it looks like this could potentially get stuck in a endless loop if cdUp fails. Also - what happens with symbolic links here? I think it needs an iteration counter where it bombs out after maxDepth iterations.

Reply:
Are you suggesting to also use the depth index as a limiter? Since one count in term of movement and the other in terms of absolute depth this could be done but I didn't want to limit the search possibilities when there are major overhauls in a folder structure. This is a balance that I find hard to achieve with two limiting elements. I'll try to restructure things.